src/go/rpk/pkg/api/admin/api_config.go

```diff
@@ -65,3 +60,76 @@ func (a *AdminAPI) SetLogLevel(name, level string, expirySeconds int) error {
 	path := fmt.Sprintf("/v1/config/log_level/%s?level=%s&expires=%d", url.PathEscape(name), level, expirySeconds)
 	return a.sendOne(http.MethodPut, path, nil, nil)
 }
+
+type ConfigPropertyItems struct {
+	Type string `json:"type"`
+}
+
+type ConfigPropertyMetadata struct {
+	Type         string              `json:"type"`
+	Description  string              `json:"description"`
+	Nullable     bool                `json:"nullable"`
+	NeedsRestart bool                `json:"needs_restart"`
+	Visibility   string              `json:"visibility"`
+	Units        string              `json:"units,omitempty"`
+	Example      string              `json:"example,omitempty"`
+	Items        ConfigPropertyItems `json:"items,omitempty"`
+}
+
+type ConfigSchema map[string]ConfigPropertyMetadata
```

Comment:
Is there a reason to not return the schema in the same api response that returns the configuration values? The API commands below look to always require grabbing both the schema and the config, should these be merged?

Reply:
It's not quite clear yet how the operator is going to consume this, but they might synchronize the schema (a copy for the GUI) and the config values separately.  Once something stateful has the schema (which they might only update after upgrades), they probably only care about the values on subsequent refreshes.